libtransmission/torrent.cc

```diff
@@ -1909,11 +1909,16 @@ void tr_torrentSetFileDLs(tr_torrent* tor, tr_file_index_t const* files, tr_file
 ****
 ***/
 
-void tr_torrent::setLabels(tr_quark const* new_labels, size_t n_labels)
+void tr_torrent::setLabels(std::vector<tr_quark> const& new_labels)
 {
     auto const lock = unique_lock();
-    auto const sorted_unique = std::set<tr_quark>{ new_labels, new_labels + n_labels };
-    this->labels = { std::begin(sorted_unique), std::end(sorted_unique) };
+    this->labels = std::vector<tr_quark>{};
+
+    for (auto l : new_labels) {
+            if (std::find(this->labels.begin(), this->labels.end(), l) == this->labels.end()) {
```

Comment:
Even though it's only a stylistic choice, please use std;:begin() and std::end() here for consistency with the rest of the codebase

Reply:
Addressed both your comments.